<?xml version="1.0"?>
<launch>
  <!-- ARGUMENTS -->
  <!--<arg name="path" default="/home/viki/catkin_ws/Bags/"/>-->
  <!--<arg name="file_1" default="sev_2016-11-24-14-48-30"/>-->
  <!-- NOTE:  tf between sensor_link and robo_link should be provided -->
  <arg name="frameid_robot" default="base_link"/>
  <!--OR: base_footprint -->
  <arg name="frameid_camera" default="stereo_camera"/>
  <!-- TF: "base_link" to "camera_link" [Rotate the camera frame.] -->
  <arg name="pi/2" value="1.5707963267948966"/>
  <arg name="optical_rotate" value="0 0 0 -$(arg pi/2) 0 -$(arg pi/2)"/>
  <node pkg="tf" type="static_transform_publisher" name="camera_base_link" args="$(arg optical_rotate) $(arg frameid_robot) $(arg frameid_camera) 30"/>
  <arg name="vehicle_rotate" value="0 0 0 0 0.13179938779906864 0"/>
  <node pkg="tf" type="static_transform_publisher" name="camera_odom_link_vehicle" args="$(arg vehicle_rotate) odom_vehicle odom 30"/>
  <node pkg="tf" type="static_transform_publisher" name="left_optical2stereo_camera" args="0 0 0 0 0 0 stereo_camera left_optical 30"/>
  <!-- PARAMETERS -->
  <param name="approx_sync" type="bool" value="true"/>
  <param name="stereo_approx_sync" type="bool" value="true"/>
  <arg name="queue_size" default="50"/>
  <!-- CHANGE FRAME_ID -->
  <node name="remap_frame_id" pkg="rustbot_bringup" type="crob_underwater_railinspect_node" output="screen"/>
  <!-- Just to uncompress images for stereo_image_rect   -->
  <node name="republish_left_slam" type="republish" pkg="image_transport" args="compressed in:=/stereo/left/image_raw raw out:=/stereo/left/image_raw"/>
  <node name="republish_right_slam" type="republish" pkg="image_transport" args="compressed in:=/stereo/right/image_raw raw out:=/stereo/right/image_raw"/>
  <!-- RESIZE -->
  <node name="ros_imresize_left" pkg="ros_imresize" type="ros_imresize" output="screen">
    <param name="camera_info" value="/stereo/left/camera_info"/>
    <param name="topic_crop" value="/stereo/left/image_raw"/>
    <param name="namespaceTopics" value="/stereo_aux/left"/>
    <param name="resize_width" value="640"/>
    <param name="resize_height" value="480"/>
    <param name="undistord" value="false"/>
  </node>
  <node name="ros_imresize_right" pkg="ros_imresize" type="ros_imresize" output="screen">
    <param name="camera_info" value="/stereo/right/camera_info"/>
    <param name="topic_crop" value="/stereo/right/image_raw"/>
    <param name="namespaceTopics" value="/stereo_aux/right"/>
    <param name="resize_width" value="640"/>
    <param name="resize_height" value="480"/>
    <param name="undistord" value="false"/>
  </node>
  <!-- END -->
  <!-- STEREO PROCESSING -->
  <group ns="stereo_camera">
    <node name="stereo_image_proc" pkg="stereo_image_proc" type="stereo_image_proc" args="--approximate 0.01" output="screen">
      <param name="approximate_sync" value="true"/>
      <param name="queue_size" value="$(arg queue_size)"/>
      <!-- <rosparam command="load" file="$(find rustbot_calibration)/calibration/stereo_image_proc.yaml"/>  -->
    </node>
  </group>
  <!-- Bags -->
  <!--<node pkg="rosbag" type="play" name="player" args="-l -clock $(arg path)$(arg file_1).bag "/>-->
  <!-- -rate 0.1 -->
</launch>
